### PR TITLE
ntpsec: update to 1.1.9

### PR DESCRIPTION
**Pull request by fhgwright**
This includes the patches for compatibility with macOS<10.13,
which can also be seen (more readably) at:

    https://gitlab.com/fhgwright/ntpsec/tree/macports_1_1_9

Switches from asciidoc to asciidoctor for building docs.

Fixes a couple of dependencies.

Fixes a bug in the default ntp.conf (more relevant now).

Adds a note regarding the new default restrictions.

Closes: https://trac.macports.org/ticket/57272

TESTED:
Tested (including building the usual set of variant combinations)
on 10.5 PPC, 10.5 i386, and 10.6-10.15 x86_64.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

Mac OS X 10.5.8 9L31a, PPC, Xcode 3.1.4 9M2809
Mac OS X 10.5.8 9L34, i386, Xcode 3.1.4 9M2809
Mac OS X 10.6.8 10K549, x86_64, Xcode 3.2.6 10M2518
Mac OS X 10.7.5 11G63, x86_64, Xcode 4.6.3 4H1503
OS X 10.8.5 12F2560, x86_64, Xcode 5.1.1 5B1008
OS X 10.9.5 13F1911, x86_64, Xcode 6.2 6C131e
OS X 10.10.5 14F2511, x86_64, Xcode 7.2 7C68
OS X 10.11.6 15G22010, x86_64, Xcode 8.1 8B62
macOS 10.12.6 16G2136, x86_64, Xcode 9.2 9C40b
macOS 10.13.6 17G13033, x86_64, Xcode 10.1 10B61
macOS 10.14.6 18G5033, x86_64, Xcode 10.3 10G8
macOS 10.15.4 19E287, x86_64, Xcode 11.5 11E608c

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [N/A] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: sysutils/ntpsec/Portfile [hunk 1]
```diff
@@ -33,8 +33,8 @@ python.versions     27
 # currently allow selecting its Python version, but since it's hard-coded
 # for 2.7, we can ignore this for now.
 
-depends_build-append    port:bison
-depends_lib-append      path:lib/libssl.dylib:openssl \
+depends_build-append    port:bison port:pkgconfig
+depends_lib-append      port:openssl \
```
Comment (cjones051073) [inline sysutils/ntpsec/Portfile:37]: Why the change in the ssl dependency ? Was there an issue with the path based form, that would have allowed compatibility with other implementations than just openssl ?

Comment (fhgwright) [reply sysutils/ntpsec/Portfile:37]: The only alternative I'm aware of is libressl, and libaes_siv (which it subsumes rather than using the port) is known to be incompatible with libressl.  It's highly unlikely that it was ever tested with anything other than OpenSSL, anyway.

Comment (kencu) [reply sysutils/ntpsec/Portfile:37]: Jeremy will come around and change all of these back to the path dependency in the end anyway, as Apple cannot apparently use openssl due to licensing issues. I've seen him do that, en-masse, many times so far.

So unless you know for certain that it can never be compatible with libressl and never will be, might as well leave it. If it needs fixing, Jeremy will swoop by and fix it someday late at night.

Comment (fhgwright) [reply sysutils/ntpsec/Portfile:38]: That makes no sense.  Why does Apple's inability to distribute OpenSSL have anything to do with whether software not distributed by Apple can depend on it?

And the "never" part also makes no sense.  If it becomes compatible with LibreSSL at some time in the future, the issue could be revisited then.  It's known not to be compatible with it **now**.  I added a comment to that effect.

The whole libaes_siv thing is somewhat interesting in its own right.  OpenSSL has included AES_SIV support for a while, but in a deficient/inefficient manner.  That's why Daniel Franke created the separate libaes_siv package, which leverages some of the OpenSSL mechanisms in a way that's not compatible with LibreSSL.  When I created the libaes_siv port (expecting it to be needed by ntpsec), I actually tried it with LibreSSL and it didn't build.  Hence:

`$ port cat libaes_siv|grep ssl`
`# NOTE: doesn't work with libressl`
`depends_lib-append  port:openssl`

That seems to have survived Jeremy so far.

I cheated and didn't retest on all platforms after adding the one comment line to the Portfile. :-)

Comment (kencu) [reply sysutils/ntpsec/Portfile:38]: Ah - not worth the argument. Go ahead and change it. Just don't be surprised if it gets changed back, like the other 100,000 times this has happened.

Comment (kencu) [reply sysutils/ntpsec/Portfile:38]: > That makes no sense. Why does Apple's inability to distribute OpenSSL have anything to do with whether software not distributed by Apple can depend on it?

Well -- now that jeremy is not around so much any more, perhaps it won't matter. He never explained it to me in detail -- I just saw it happen over and over. I think he is legally unable to look at the openssl code -- perhaps even to work with it -- as part of his job with Apple. 

Jeremy uses (or perhaps "used" might be a better term) many of the MacPorts libraries for his projects, sometimes personal (XQuartz) and at least in the older days, sometimes related to Apple things. I think that is why he was always so careful about changing every port in MacPorts to always be able to build with libressl, and he would go around and make changes in various ports to make them compatible with libressl if they were not.

In this particular port, I see he never changed it -- it was always a path dep, right from the beginning. But there are many other examples where it was changed to a path dep.

Now we never used to do things that would purposefully p*ss Jeremy off -- after all, what other open source project in Darwin-land had their own personal Apple engineer overseeing everything? But he's not around, so maybe it doesn't matter any more. 

I certainly don't care one way or the other as it is totally meaningless to me. 


> I cheated and didn't retest on all platforms after adding the one comment line to the Portfile. :-)

You do a lot of testing on the various systems, as do I -- that's much appreciated, rather than the "push it and watch for wreckage" approach :>
